algoliasearch/types_rule.go

```diff
@@ -0,0 +1,93 @@
+package algoliasearch
+
+type Rule struct {
+	ObjectID    string          `json:"objectID,omitempty"`
+	Condition   RuleCondition   `json:"condition"`
+	Consequence RuleConsequence `json:"consequence"`
+	Description string          `json:"description,omitempty"`
+}
+
+// RuleCondition is the part of an Algolia Rule which describes the condition
+// for the rule. The `Context` and `Priority` are optional, hence, they will
+// get ignored if an empty string or a 0 values are respectively set to them.
+type RuleCondition struct {
+	Anchoring RulePatternAnchoring `json:"anchoring"`
+	Pattern   string               `json:"pattern"`
+	Context   string               `json:"context,omitempty"`
+	Priority  int                  `json:"priority,omitempty"`
```

Comment:
Sorry, maybe it wasn't obvious in the doc, but `priority` is actually not supported. It was left out of the scope until further notice. => Please remove it.